doc/src/compute_mliap.rst

```diff
@@ -0,0 +1,170 @@
+.. index:: compute mliap
+
+compute mliap command
+=====================
+
+Syntax
+""""""
+
+.. code-block:: LAMMPS
+
+   compute ID group-ID mliap ... keyword values ...
+
+* ID, group-ID are documented in :doc:`compute <compute>` command
+* mliap = style name of this compute command
+* two keyword/value pairs must be appended
+* keyword = *model* or *descriptor*
+
+  .. parsed-literal::
+
+       *model* values = style Nelems Nparams
+         style = *linear* or *quadratic*
+         Nelems = number of elements
+         Nparams = number of parameters per element 
+       *descriptor* values = style filename
+         style = *sna*
+         filename = name of file containing descriptor definitions
+
+Examples
+""""""""
+
+.. code-block:: LAMMPS
+
+   compute mliap model linear 2 31 descriptor sna Ta06A.mliap.descriptor
+
+Description
+"""""""""""
+
+Compute style *mliap* provides a general interface to the gradient
+of machine-learning interatomic potentials w.r.t. model parameters. 
+It is used primarily for calculating the gradient of energy, force, and
+stress components w.r.t. model parameters, which is useful when training
+:doc:`mliap pair_style <pair_mliap>` models to match target data.
+It provides separate 
+definitions of the interatomic potential functional form (*model*)
+and the geometric quantities that characterize the atomic positions
+(*descriptor*). By defining *model* and *descriptor* separately, 
+it is possible to use many different models with a given descriptor,
+or many different descriptors with a given model. Currently, the 
+compute supports just two models, *linear* and *quadratic*,
+and one descriptor, *sna*, the SNAP descriptor used by 
+:doc:`pair_style snap <pair_snap>`, including the linear, quadratic,
+and chem variants. Work is currently underway to extend
+the interface to handle neural network energy models,
+and it is also straightforward to add new descriptor styles.
+
+The compute *mliap* command must be followed by two keywords
+*model* and *descriptor* in either order. 
+
+The *model* keyword is followed by a model style, currently limited to
+either *linear* or *quadratic*. In both cases,
+this is followed by two arguments. *nelems* is the number of elements.
+It must be equal to the number of LAMMPS atom types. *nparams*
+is the number of parameters per element for this model i.e.
+the number of parameter gradients for each element. Note these definitions
+are identical to those of *nelems* and *nparams* in the 
+:doc:`pair_style mliap <pair_mliap>` model file.
+ 
+The *descriptor* keyword is followed by a descriptor style, and additional arguments.
+Currently the only descriptor style is *sna*, indicating the bispectrum component 
+descriptors used by the Spectral Neighbor Analysis Potential (SNAP) potentials of 
+:doc:`pair_style snap <pair_snap>`.
+The \'p\' in SNAP is dropped, because keywords that match pair_styles are silently stripped 
```

Comment:
@athomps do you know why this is? should we try to do something about it? as you may recall from fixing the pair style hybrid error message, we have now better facilities to do keyword matching (and that uses only one of them).

Reply:
Yes, my original diagnosis was wrong. The keyword snap is not silently stripped, rather it is matched to a pair style. This causes PairMLIAP::settings() to be called with a truncated argument list.  Ironically, this would have worked with the old pair hybrid code, before it was fixed.  I don't see a need to fix this new issue and I am okay with removing the comment "The \'p\' in SNAP is dropped, because keywords that match pair_styles are silently stripped."